aten/src/ATen/native/cuda/foreach/MultiTensorApply.cuh

```diff
@@ -0,0 +1,75 @@
+#include <torch/torch.h>
+#include <torch/cuda.h>
+#include <ATen/ATen.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/native/cuda/foreach/Utils.cuh>
+
```

Comment:
Was this file based off of https://github.com/NVIDIA/apex/blob/master/csrc/multi_tensor_apply.cuh ? It looks pretty difference, is there a description of the algorithmic differences?